rsconnect/api.py

```diff
@@ -1254,9 +1270,23 @@ def prepare_deploy(self, app_id: typing.Optional[int], app_name: str, bundle_siz
 
         if app_id is None:
             application = self._rstudio_client.create_application(account["id"], app_name)
+            self._server.handle_bad_response(application)
+            if visibility is not None:
+                property_update = self._rstudio_client.update_application_property(
+                    application["id"], "application.visibility", visibility
+                )
+                self._server.handle_bad_response(property_update)
+
         else:
             application = self._rstudio_client.get_application(app_id)
-        self._server.handle_bad_response(application)
+            self._server.handle_bad_response(application)
+
+            if visibility is not None:
+                if visibility != application["deployment"]["properties"]["application.visibility"]:
+                    self._rstudio_client.update_application_property(
+                        application["id"], "application.visibility", visibility
+                    )
```

Comment:
Does this need to call `handle_bad_response`?

Reply:
Yes :) thanks for catching that.